From 862ac204d5d09f6d53e9b495076401352abe1eef Mon Sep 17 00:00:00 2001
From: Grace Cham <hscham@chromium.org>
Date: Wed, 15 Jun 2022 23:23:05 +0900
Subject: [PATCH] Remove ICU dependency for base::Time::Explode

ICU dependency was introduced for Explode implementation to tackle Y2038
problem but it signigicantly increases binary size. Removed as Y2038
boards will be EOL by then.

---
 base/time/time_exploded_posix.cc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/base/time/time_exploded_posix.cc b/base/time/time_exploded_posix.cc
index 45327a89b77c..650738d5bcec 100644
--- a/base/time/time_exploded_posix.cc
+++ b/base/time/time_exploded_posix.cc
@@ -128,12 +128,14 @@ void Time::Explode(bool is_local, Exploded* exploded) const {
 
   // For systems with a Y2038 problem, use ICU as the Explode() implementation.
   if (sizeof(SysTime) < 8) {
+#if 0 // Not to introduce icu dependency for now. Y2038 boards will be EOL by then.
 // TODO(b/167763382) Find an alternate solution for Chromecast devices, since
 // adding the icui18n dep significantly increases the binary size.
 #if !BUILDFLAG(IS_CASTOS) && !BUILDFLAG(IS_CAST_ANDROID)
     ExplodeUsingIcu(millis_since_unix_epoch, is_local, exploded);
     return;
 #endif  // !BUILDFLAG(IS_CASTOS) && !BUILDFLAG(IS_CAST_ANDROID)
+#endif
   }
 
   // Split the |millis_since_unix_epoch| into separate seconds and millisecond
-- 
2.37.1.359.gd136c6c3e2-goog

